### ISSUE TITLE
[Bug] WebKitWebProcess causes high CPU usage on linux

### ISSUE THREAD

**Issue by WizardOfCodez** [wailsapp/wails#577]
**Description**
As described on issue https://github.com/rule110-io/surge/issues/40 WebKitWebProcess takes up a lot of CPU for rendering our app

It seems like that is caused by a bug in WebKitGTP which is described here: https://github.com/aperezdc/revolt/issues/96#issuecomment-581888793

Is it possible to fix this?

**Comment by leaanthony** [comment]
Happy to accept a PR for this! Is there any reason this shouldn't be default behaviour? Should we make it configurable? Did you try linking against the later GTK as suggested?

**Comment by CrackDavid** [comment]
Already using the latest webkit2GTK here, so no problems with updating. Is there any way to set  ``websettings.set_hardware_acceleration_policy(WebKit2.HardwareAccelerationPolicy.ALWAYS)`` with build flags?

**Comment by leaanthony** [comment]
It feels like a runtime issue but maybe we could use build tags to include code at compile time to set it?

**Comment by CrackDavid** [comment]
I tried to play a little bit around but no compile flags I found. Also tried to find a way to set setting in the webview-go implementation but without luck.

**Comment by leaanthony** [comment]
I'll try and have a look at it over the "holidays" - if you can call doing anything but resting holidays :joy:

**Comment by leaanthony** [comment]
@CrackDavid - If you install the latest develop branch and make sure you use `replace` in your `go.mod` to point to it, then compiling your app in debug mode should set this setting.

I've tested locally here and it compiles fine but I'm unable to test the high CPU usage issue so please let me know how you get on. This is the relevant commit: https://github.com/wailsapp/wails/commit/9cd5ad69ce16d7cf610dab1e32d7c55e6baa60a5#diff-66741ae7986bce2f3e03694005c862de7e520804fb0acd0febf399610dc4d14aR367

Basically, if you can right click and "inspect element" then this fix should be in place. Let me know how you get on 👍

**Comment by CrackDavid** [comment]
@leaanthony - I did apply the changes but nothing really changed so the flag should not be the error. 
Does "can't test it" mean that on a boilerplate project everything is fine so it could be an issue with our app or do you got no unix-VM at hand?

**Comment by leaanthony** [comment]
I meant i was short on time. I'll try to find time today. Try adding a `printf("applying setting\n");` before the updated line to make sure it's being applied. If that is printing and you still have issues, try profiling the application to find out where the CPU is being chomped.

**Comment by CrackDavid** [comment]
I already made sure the changes are applied, since it is very late here I will try to get more informations tomorrow.
